packages/core/admin/admin/src/hooks/useSettingsForm/tests/index.test.js

```diff
@@ -0,0 +1,125 @@
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useNotification: jest.fn().mockImplementation(() => toggleNotification),
+  useOverlayBlocker: () => ({ lockApp: jest.fn(), unlockApp: jest.fn() }),
+}));
+
+jest.mock('../../../utils', () => ({
+  ...jest.requireActual('../../../utils'),
+  checkFormValidity: () => (null),
+}));
+
+jest.mock('lodash/omit');
+
+const handlers = [
+  rest.put('*/providers/options', (req, res, ctx) =>
+    res(
+      ctx.status(200),
+      ctx.json({
+        data: {
+          autoRegister: false,
+          defaultRole: "1",
+          ssoLockedRoles: ["1","2","3"]
+        }
+      })
+    )
+  ),
+  rest.get('*/providers/options', (req, res, ctx) =>
+    res(
+      ctx.status(200),
+      ctx.json({
+        data: {
+          autoRegister: false,
+          defaultRole: "1",
+          ssoLockedRoles: ["1","2"]
+        }
+      })
+    )
+  )
+];
+
+const server = setupServer(...handlers);
+
+const setup = (...args) =>
+  renderHook(() => useSettingsForm(...args), {
+    wrapper({ children }) {
+      return (
+        <IntlProvider locale="en" messages={{}}>
+          {children}
+        </IntlProvider>
+      );
+    }
+  })
+
+describe('useSettingsForm', () => {
+  beforeAll(() => {
+    server.listen();
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+  test('fetches all the providers options', async () => {
+    const { result, waitFor } = setup('/admin/providers/options', {
+      validate: jest.fn()
+    }, jest.fn(), ['autoRegister', 'defaultRole', 'ssoLockedRoles'] );
+
+    expect(result.current[0].isLoading).toBe(true);
+    expect(result.current[0].formErrors).toStrictEqual({});
+    expect(result.current[0].initialData).toStrictEqual({});
+    expect(result.current[0].modifiedData).toStrictEqual({});
+    expect(result.current[0].showHeaderButtonLoader).toBeFalsy();
+    expect(result.current[0].showHeaderLoader).toBeTruthy();
+
+    await waitFor(() => expect(result.current[0].isLoading).toBe(false));
+    
+    expect(result.current[0].formErrors).toStrictEqual({});
+    expect(result.current[0].initialData).toStrictEqual(
+      expect.objectContaining({
+        autoRegister: false,
+        defaultRole: "1",
+        ssoLockedRoles: ["1","2"]
+      })
+    );
+
+    expect(result.current[0].modifiedData).toStrictEqual(
+      expect.objectContaining({
+        autoRegister: false,
+        defaultRole: "1",
+        ssoLockedRoles: ["1","2"]
+      })
+    );
+
+    expect(result.current[0].showHeaderButtonLoader).toBeFalsy();
+    expect(result.current[0].showHeaderLoader).toBeFalsy();
+  });
+
+  test('submit new providers options with duplications', async () => {
+    const cbSucc = jest.fn();
+    const ssoLockedRolesWithDuplications = [ '1', '2', '2', '3' ];
+    omit.mockReturnValueOnce({ 
+      autoRegister: false,
+      defaultRole: '1',
+      ssoLockedRoles: ssoLockedRolesWithDuplications 
+    });
```

Comment:
I'm not sure what you are trying to achieve here, but it is very brittle to mock hook internals. Your tests shouldn't be dependent on this, because it makes the test pointless if someone can not refactor it without changing the tests.

I think you should find a better way to test that ...

Reply:
I think you could overwrite a route for a single test using:

```
const server = createServer(handlers);

test('special test', () => {
  server.use(
    rest.get('/route', (req, res) => res.once(ctx.json({})))
  )
})
```

https://mswjs.io/docs/api/setup-server/use

?

Reply:
it works, thanks